vendors/nordic/boards/nrf52840-dk/ports/ble/iot_ble_hal_gap.c

```diff
@@ -838,14 +839,16 @@ ret_code_t prvBTAdvDataConvert( ble_advdata_t * xAdvData,
             else
             {
                 /* extract company_identifier from input manufacturer data
-                because nordic sdk has separate field for company_identifier */
-                for (uint8_t i = 0; i < iot_ble_hal_gapADVERTISING_COMPANY_ID_SIZE; i++) {
-                  companyId = companyId | ((uint16_t) pcManufacturerData[i] << (i * 8));
+                 * because nordic sdk has separate field for company_identifier */
+                for( uint8_t i = 0; i < iot_ble_hal_gapADVERTISING_COMPANY_ID_SIZE; i++ )
```

Comment:
Would add comments about the size of the company ID string (I believe it's 4 bytes based on the rest of the code) and the size of `companyId` since it's not encoded in that variable name.

Reply:
Added comment regarding the company identifier field and the variable.